src/main/java/ch/geowerkstatt/ilivalidator/extensions/functions/IsInsideExternalDatasetIoxPlugin.java

```diff
@@ -0,0 +1,406 @@
+
+        if (argDatasetName.isUndefined() || argObjects.isUndefined() || argTestObject.isUndefined() || argTestObjectgeometry.isUndefined()) {
+            return Value.createSkipEvaluation();
+        }
+
+        try {
+            String transferIds = argObjects.getValue();
+            String[] splitTransferIds = transferIds.split("\\s*,\\s*");
+            String datasetName = argDatasetName.getValue();
+
+            ValidAreaKey key = new ValidAreaKey(datasetName, splitTransferIds);
+            Geometry validArea = validAreaCache.computeIfAbsent(key, this::getValidArea);
+
+            if (validArea == null) {
+                return Value.createUndefined();
+            }
+
+            String testObjectgeometryAttribute = argTestObjectgeometry.getValue();
+
+            boolean allInsideValidArea = argTestObject.getComplexObjects().stream()
+                    .flatMap(obj -> getAttributes(obj, testObjectgeometryAttribute).stream())
+                    .map(a -> logExceptionAsWarning(() -> geometry2JtsOrNull(a)))
+                    .filter(Objects::nonNull)
+                    .allMatch(validArea::contains);
+
+            return new Value(allInsideValidArea);
+        } catch (IllegalStateException e) {
+            logger.addEvent(logger.logErrorMsg(MessageFormat.format("{0}: Unable to evaluate {1}. {2}", usageScope, this.getQualifiedIliName(), e.getLocalizedMessage())));
+            return Value.createUndefined();
+        }
+    }
+
+    /**
+     * Retrieve the geometries from external files according to the specified key.
+     *
+     * @param key The key that specifies which geometries to retrieve.
+     * @return The union of all found geometries.
+     */
+    private Geometry getValidArea(ValidAreaKey key) {
+        int firstPoint = key.datasetName.indexOf('.');
+        int lastPoint = key.datasetName.lastIndexOf('.');
+        if (firstPoint == -1 || lastPoint == -1) {
+            throw new IllegalStateException(MessageFormat.format("Expected qualified attribute name but got <{0}>.", key.datasetName));
+        }
+
+        String model = key.datasetName.substring(0, firstPoint);
+        String attribute = key.datasetName.substring(lastPoint + 1);
+        String qualifiedClass = key.datasetName.substring(0, lastPoint);
+
+        Collection<URL> urls = getUrlsForModel(model);
+        if (urls == null) {
+            throw new IllegalStateException(MessageFormat.format("Could not find Transferfile containing model <{0}>.", model));
+        }
+
+        Polygon[] polygons = urls.stream()
+                .map(url -> logExceptionAsWarning(url::openStream))
+                .flatMap(xtfStream -> getObjectsFromXTF(xtfStream, qualifiedClass, key.transferIds).stream())
+                .flatMap(obj -> getAttributes(obj, attribute).stream())
+                .map(a -> logExceptionAsWarning(() -> Iox2jtsext.surface2JTS(a, 0.0, new OutParam<Boolean>(), logger, 0.0, "warning")))
```

Comment:
Q: Kann JTS noch keine Arcs? Können wir mit 0.0 überhaupt Stroken? 
PP: Wenn wir Arcs stroken müssen, dann sollte dies in der Funktionsbeschreibung auftauchten.

Reply:
Ich nehme an, das du dich auf den Parameter `strokeP` beziehst. Dieser wird in `surface2JTS` an `polyline2JTS` weitergereicht aber dort nicht verwendet 🤷